OJP/OJP_Trips.xsd

```diff
@@ -117,6 +130,31 @@
 					<xs:documentation>For each mode in this list a separate monomodal trip shall be found - in addition to inter-modal solutions.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="IncludeAlternativeOptions" type="xs:boolean" default="false" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Whether alternative options should be presented as well. Mainly important for dominated journeys or in the case of ContinuousLegs the second-best route. Should be optimised for the user expectance (see. e.g https://theses.hal.science/tel-01848737). However, what the alternative options are may vary widely depending on the optimisation methods and filters.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="IncludePrivateLines" type="xs:boolean" minOccurs="0" default="false">
+				<xs:annotation>
+					<xs:documentation>There might exist lines that have special restrictions and are not available to the public. E.g. school buses, company shuttles. If this flag is set, then the private lines are considered by the router. The results are marked as private with a hint to the restriction.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="IncludeTouristicOffers" type="xs:boolean" minOccurs="0" default="false">
+				<xs:annotation>
+					<xs:documentation>With this option set, legs with ACCESS MODE like ski are also searched. If this is not set, then skiLift are not accessible. The OptimisationMethod scenic is more about the view and other aspects.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="PassengerCategory" type="PassengerCategoryEnumeration" minOccurs="0" maxOccurs="unbounded">
+				<xs:annotation>
+					<xs:documentation>Sequence of all passenger categories, for which this search shall be conducted. In rare cases additional offerings may be available (e.g., demand responsive service with certain areas being reserved for seniors).</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="MainTravelMode" type="ifopt:AccessModesEnumeration" default="foot" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>The MainTravelMode defines whether the trip planner works in PT mode or includes carTransportRail and the like. The only mode that activates the inclusion of this is "car".</xs:documentation>
+				</xs:annotation>
+			</xs:element>
```

Comment:
I'm sorry for again raising questions because I still don't sufficiently understand the aim.

**_MainTravelMode_** car (would also work for motorcycle). What is expected from the routing algorithm?
	1. Variant 1: With the exception of the initial and final access legs, the whole trip is accessible for the car. 
	2. Variant 2: A substantial part of the trip, consisting of a connected sequence of legs, is accessible for the car. 
	3. Variant 3: The planning is done like for a normal passenger, but with additional services unlocked that might allow for additional or faster solutions.

In variants 1 and 2, the routing algorithm searches for trips involving at least one PT leg and minimising the distance covered by the continuous legs. (Or is that too complicated because of the combination of PT and continuous planning?)
In variant 3, MainTravelMode car would just unlock additional services, the algorithm would not care whether the trip is feasible or somehow optimal for an automobilist.

Both 1 and 2 would be able to find the car transport rail Goppenstein-Kandersteg, or a ferry allowing to shorten driving distance.
Variant 3 might also find them, but only "by chance", if they appear in a pure PT / time table search.
Variant 1 would be able to find a nightly car transport rail service Basel-Hamburg allowing to shorten driving distance. 
Variant 2 might limit the nightly car transport rail to Basel-Freiburg and propose to continue by a much faster ICE, leaving the car in Freiburg. This I would consider a strong argument against variant 2. 
Variant 3 would be able to consider the nightly car transport service, but probably would abandon it as it is very slow.
Variants 2 and 3, but not 1, could find a trip involving car transport rail Goppenstein-Kandersteg and a drag lift to a mountain top given MainTransportMode car and IncludeTouristicOffers.


**_IncludeTouristicOffers_** (and access modes as proposed previously)  -  unlocks additional services

Without a MainTransportMode like car or motorcycle ->  a pure PT search having access to unlocked additional services (like variant 3)

If access mode car still existed, this would allow for finding a trip involving car transport rail Goppenstein-Kandersteg and a drag lift to a mountain top given access mode car and IncludeTouristicOffers. Thus circumventing the limitation of variant 1 mentioned above.


Additional considerations: 
Transfer legs: variant 1 would not include any transfer legs because _paths between PT legs have to be continuous legs_ (car)
Transfer legs without a main mode like car or motorcycle: should the algorithm try to replace transfer legs with continuous legs if an adjacent PT leg is accessible by car? 
How about bicycles? 

**Preliminary conclusion:**

The **most useful solution** seems to me: variant 1 for MainTransportMode car and motorcycle, And to keep access modes. For me variant 1 feels like invoking an optimisation method aiming to propose attractive car transport offers.

The **simplest solution** seems to me, but of very limited use: variant 3, but for clarity I would propose to rename _MainTransportMode_ to _IncludeVehicleTransportOffers_ (for car, motorcycle, bicycle).

Reply:
The choice construct to separate behaviors (1) and (3) seems like a good solution to me.

I think keeping MainTransportMode is better if/as the intention is variant (1).

I think IncludeTouristicOffers is also a good solution as it allows to simply unlock a bunch of services and it is the responsability ot the user to determine whether he can use it (necessary equipment, consistency of trip). But how is he informed about that? There is the _AccessModeList_ for places, but I don't see how this could tell something about trip legs.

How about motorcycles? If they're used to be considered a "kind of car", _car_ may be sufficient in MainTransportMode to activate the behavior (1), otherwise I think _motorcycle_ should also be valid/activating.

How about bicycles? I think the best behavior would be: if the user selects _BikeTransport_ (TripMobilityFilter) then the filter should  not only restrict the search, but unlock services for which bicycle is an access mode.